app/src/main/java/com/example/myapplication/data/retrofit/AppModule.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.myapplication.data.retrofit
+
+import com.example.myapplication.data.api.UserApi
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class AppModule {
+    @Provides
+    @Singleton
+    fun provideClient():OkHttpClient=OkHttpClient.Builder()
+        .addInterceptor(createInterceptor())
+        .build()
+
+    private fun createInterceptor(): HttpLoggingInterceptor {
+        val interceptor = HttpLoggingInterceptor()
+        interceptor.level = HttpLoggingInterceptor.Level.BODY
+        return interceptor
+    }
+
+    @Provides
+    @Singleton
+    fun provideRetrofit(client: OkHttpClient): Retrofit = Retrofit.Builder()
+        .client(client)
+        .addConverterFactory(GsonConverterFactory.create())
+        .baseUrl("http://178.63.9.114:7777/api/")
```

Comment:
try storing this base url as a gradle constant, useful for having multiple product flavors